### DISCUSSION TITLE
label on crosshair does not update

### DISCUSSION THREAD

**Discussion by Eekkow** [Live-Charts/LiveCharts2#1655] [Q&A]
Hello,  i have a question: when we update or redraw the x-axis, the labels change, but the label on the crosshair does not. Is this a bug, or is it the intended behavior?

For instance, if the x-axis labels are initially 1, 2, 3, 4, 5, 6, 7, 8, 9, and the crosshair label shows '4,' when we change the data on the page, the x-axis labels are updated to 'Jan, Feb, March, April,' etc. However, the crosshair label still shows '4,' when we would expect it to update to match the new x-axis labels, such as 'April.' How can we ensure the crosshair label updates along with the x-axis labels?"

**Comment by TheSimplyCode** [comment]
I had the same problem, my current solution is to reload the XAxes and the YAxes, sometimes several times, then the label is correct again. I have the feeling that the data must be displayed first, only then can a reload be made so that the label is then really adopted.
I have done something like that:
```cs
    public void InitChartAxis()
    {
        Frame = AxisFactory.DefaultFrame();

        XAxes = new ObservableCollection<Axis>()
        {
           AxisFactory.CreateDefaultAxis("X", new Padding(5, 15, 5, 5))
        };

        YAxes = new ObservableCollection<Axis>()
        {
          AxisFactory.CreateDefaultAxis("Y", new Padding(5, 0, 15, 0))
        };
    }
```
I have not yet found the right event to trigger the reload. At the moment I have made it so that I have to click in the chart and then the reload event is triggered. From my point of view this is a `BUG`.

![image](https://github.com/user-attachments/assets/2aef3d04-234f-46b7-89e5-e2ccf731caf7)
